docs/server/ResourceServer.md

```diff
@@ -0,0 +1,141 @@
+# Resource Server
+
+Gateway includes a Laravel authentication guard, user provider, and middleware for easily configuring applications to work with OAuth tokens. This allows your application to verify tokens created by [Northstar](https://github.com/dosomething/northstar) (or another OpenID Connect server).
+
+## Installation
+
+To start, let's configure Gateway's guard & user provider. In your `config/auth.php`:
+
+```php
+<?php
+
+return [
+    // ...
+
+    'guards' => [
+        // ...
+
+        'api' => [
+            'driver' => 'gateway',
+            'provider' => 'northstar',
+        ],
+    ],
+
+    'providers' => [
+        'northstar' => [
+            'driver' => 'gateway',
+            'url' => env('NORTHSTAR_URL'),
+            'key' => storage_path('keys/public.key'),
```

Comment:
Note: These two are actually lies for now (it currently reads them from `config/services.php`). I'm going to move them in a follow-up PR since they make a lot more sense here.